### DISCUSSION TITLE
Why there is few deno frameworks support web worker?

### DISCUSSION THREAD

**Discussion by c608345** [denoland/deno#15554] [General]
https://github.com/honojs/hono/issues/470

The framework author seems refuse web worker without reasons.

**Comment by kitsonk** [comment]
Deno doesn't support service workers.  You mean web workers?

Really that is the question for the framework authors.

Hono isn't specifically a Deno framework, so there are the considerations of other platforms.  Typically cloud based workers don't support additional "owned" workers and achieve scalability by multiple instances, so it might not ever make sense for something like Hono.

Ultimately, web workers have a lot of limited usability, where you are doing a lot of processing but there is little data transfer between the main worker and the web worker, especially when it comes to processing network requests, there is almost 0 benefit, because the runtime is really only handling requests with a single worker in mind.  In most cases you really want to look at making the main thread as performant as possible, blocking as little as possible.

**Comment by c608345** [comment]
@kitsonk 
In server environments not cloud functions. How deno frameworks use multi-core? When run benchmark, QPS is low and CPU is not full loaded.
